lib/src/err/mod.rs

```diff
@@ -633,7 +633,16 @@ pub enum Error {
 	/// This should be used extremely sporadically, since we lose the type of error as a consequence
 	/// There will be times when it is useful, such as with unusual type conversion errors
 	#[error("Internal database error: {0}")]
-	Internal(String),
+	Internal(InternalCause),
+
+	/// Internal server error related to context
+	/// A classification of internal error, related directly to context
+	#[error("Internal database error due to context: {0}")]
+	InternalContextError(ContextCause),
```

Comment:
Since we are using Bincode, the position of an enum variant is important. New enum variants should always go to the bottom of an enum, otherwise it will be a breaking change.